src/main/java/io/jenkins/plugins/appcenter/task/internal/UploadAppToResourceTask.java

```diff
@@ -84,11 +85,39 @@
         final String pathToDebugSymbols = request.pathToDebugSymbols;
         final String symbolUploadUrl = requireNonNull(request.symbolUploadUrl, "symbolUploadUrl cannot be null");
 
-        log("Uploading symbols to resource.");
+        final File file = remoteFileUtils.getRemoteFile(pathToDebugSymbols);
+        if (file.length() > MAX_NON_CHUNKED_UPLOAD_SIZE) {
+            return uploadSymbolsChunked(request, symbolUploadUrl, file);
+        } else {
+            return uploadSymbolsComplete(request, symbolUploadUrl, file);
+        }
+    }
+
+    @Nonnull
+    private CompletableFuture<UploadRequest> uploadSymbolsChunked(@Nonnull UploadRequest request, @Nonnull String symbolUploadUrl, @Nonnull File file) {
+        log("Uploading symbols to resource chunked.");
 
         final CompletableFuture<UploadRequest> future = new CompletableFuture<>();
 
-        final File file = new File(filePath.child(pathToDebugSymbols).getRemote());
+        CompletableFuture.runAsync(() -> factory.createBlobUploadService(symbolUploadUrl).uploadFromFile(file.getPath(), true))
```

Comment:
I did a bit of research again on `supplyAsync` and `runAsync` and if we do not return a value i.e. the function returns `null` then we can use `runAsync` like this.